oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/IndexUpdate.java

```diff
@@ -330,9 +334,14 @@ private void collectIndexEditors(NodeBuilder definitions,
                     // then we don't need to handle missing handler
                     if (definition.hasProperty(ASYNC_PROPERTY_NAME) && rootState.async == null) {
                         if (!TYPE_DISABLED.equals(type)) {
-                            log.warn("Missing provider for nrt/sync index: {} (rootState.async: {}). " +
-                                    "Please note, it means that index data should be trusted only after this index " +
-                                    "is processed in an async indexing cycle.", definition, rootState.async);
+                            long now = System.currentTimeMillis();
+                            if (now > lastMissingProviderMessageTime + 60 * 1000) {
+                                lastMissingProviderMessageTime = now;
+                                log.warn("Missing provider for nrt/sync index: {} (rootState.async: {}). " +
+                                        "Please note, it means that index data should be trusted only after this index " +
+                                        "is processed in an async indexing cycle. " +
+                                        "This message is silenced for one minute.", definition, rootState.async);
```

Comment:
the placeholder in 

```Missing provider for nrt/sync index: {}```

is replaced with the variable `definition` which is an instance of NodeBuilder. Should we instead pass `indexPath` defined at line 312?